src/Features/WcPayPromotion/WCPaymentGatewayPreInstallWCPayPromotion.php

```diff
@@ -35,5 +35,24 @@ public function __construct() {
 
 		// Get setting values.
 		$this->enabled = false;
+
+		// Load the settings.
+		$this->init_form_fields();
+		$this->init_settings();
+	}
+
+	/**
+	 * Initialise Gateway Settings Form Fields.
+	 */
+	public function init_form_fields() {
+		$this->form_fields = array(
```

Comment:
Nice! I wouldn't of thought of this, this is a clever way to keep track of the dismissed state 👍